### ISSUE TITLE
VaRadio component selection not changing with arrow keys - 508 audit finding (High severity)

### ISSUE THREAD

**Issue by hdjustice** [department-of-veterans-affairs/vets-design-system-documentation#3928]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

- 10-10 Health Apps team received a 508 Accessibility audit finding on the 10-10cg Facilities search feature, specifically the radio button selection using keyboard navigation
- This seems to be an issue built into the `VaRadio` component
- This is considered a **HIGH** severity 508 Accessibility audit finding

> **Website interferes with intended keyboard functionality when JAWS is running**.
> Defect 1 | High | Created:
> 502.2.2 No Disruption of Accessibility Features | Software shall not disrupt platform features that are defined in the platform documentation as accessibility features.
> Third Page.
> (New 03/18/2025) When assistive technology is running, the user is unable to change the selection of a radio button group using standard navigation. When a user makes a selection in a radio button group, the expectation is that the user may change the selection using the arrow keys. However, when JAWS is running, and the user attempts to change the selection of the “Select the Veteran’s VA medical center or clinic” radio button group using the arrow keys, the virtual cursor is moved, but not the selection. In order to change the selection, the user must press the space bar

![Image](https://github.com/user-attachments/assets/e75c339a-7dcf-4b68-be19-51a89dd704cd)

## What I expected to happen

- When a user makes a selection in a radio button group, the expectation is that the user may change the selection using the arrow keys. 

## Reproducing

- CSS Library version: Unknown
- Device: (e.g. iPhone 8, Macbook) Unknown
- Browser: (e.g. Firefox, IE 11) Unknown
 
Steps to reproduce:

1. Type in city, state or postal code and select Search button
2. Navigate to list of facilities in the search results
3. Select a facility
4. Use arrow keys to move up and down the facility search results list, which should also change the selection

## Urgency

How urgent is this request? Please select the appropriate option below and/or provide details

- [ ] This bug is blocking work currently in progress
- [ ] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [x] This bug is not blocking any work
- [x] Other

### Details
- We have already released this functionality into production, but concerned about the severity of the finding

**Comment by coforma-jamie** [comment]
@Mottie and I did some digging. We think we figured out what the problem is:

## About forms mode
- In JAWS, there are two modes. "Browse mode" allows you to arrow through individual elements of a web page and announce their contents, "forms mode" lets you interact with form inputs.
- There are different ways to trigger "forms mode," depending on your settings. With the default settings, you trigger it by putting focus on the form input, then pressing SPACE or ENTER (depending on the input type) to enter/activate the field.

Here's an example of JAWS entering forms mode - note the lower-pitched tone for browse mode, and the higher-pitched tone for forms mode:

https://github.com/user-attachments/assets/749603eb-8980-416a-88c5-ff183c055035

**This is standard behavior.**

## What's supposed to happen with `input type="radio"`
When a JAWS user arrows down the page and enters a VADS radio button group, focus moves to the first input (the first radio button). 

When they press DOWN, focus moves to the second input, then the third. If the user presses ENTER on any of those inputs, forms mode is activated and the radio button is checked:

https://github.com/user-attachments/assets/519741b3-54a8-4009-a290-4aad7b58d99c


## What's happening with `<va-radio>`
When a JAWS user arrows down the page and enters a VADS radio button group, focus moves to the first input (the first radio button). 

**If they keep pressing DOWN**, focus moves to the first **label**, then the second radio button, then the second label, etc. **This is not standard behavior.**

> **Note:** The input and label both have the same accessible name! If you have a VADS radio button like this - `() Yes` - the input will announce `Yes, radio button, unchecked`, and the label will announce `Yes`.

If the user presses ENTER when focus is on the label - not the input - it "checks" the radio button, but forms mode doesn't come on. 

You can see that in this second example - I pressed DOWN, put focus on the label, and pressed ENTER. JAWS didn't enter forms mode. **Since JAWS was still in browse mode, pressing UP and DOWN within the radio button group didn't correctly navigate through the radio buttons.**

https://github.com/user-attachments/assets/7c65d5b8-3b74-4a1c-9065-ae1af2b5ca78

## The issue
We believe that the issue is with [this code](https://github.com/department-of-veterans-affairs/component-library/blob/main/packages/web-components/src/components/va-radio/va-radio.tsx#L135-L170). There are custom keystrokes defined for `va-radio`, but we don't think there needs to be, since `va-radio-option` renders to `input type="radio"`.

We think that if this code is removed, the problem may be solved.

**Comment by caw310** [comment]
Please add your estimate of 1, 2, 3, 5, 8 or 13
@Andrew565 
@ataker 
@harshil1793 
@it-harrison 
@jamigibbs 
@micahchiang 
@powellkerry 
@rmessina1010 
@rsmithadhoc

**Comment by caw310** [comment]
This may carry over as Ryan is out and we need his help testing Jaws.

**Comment by caw310** [comment]
This will also carry over because this is a bigger issue than indicated in this ticket. There are implications for va-radio and va-checkbox.

**Comment by rsmithadhoc** [comment]
Hi @coforma-jamie,

After a lot of testing, I believe this has to do with the additional description in the radio label, and I was only able to recreate this using forms manual mode in JAWS. 

With the additional label, using the down arrow will:
1. Read the radio button
2. Read the label as separate text on the page.
For #1, pressing Enter went into forms mode. For #2, pressing Enter did not enter forms mode, thus the selection could not be changed. With just a plain radio label, this only announced the label once, so this was not an issue.

I was able to recreate the issue with only the USWDS markup as well. I tried other ways of adding the additional description to the radio label (it currently adds a `<span>`), but the result was the same.

So my questions are:
- I could not get this to happen with JAWS' default automatic forms mode. Are you able to recreate the issue with that mode too? I typically test with the screen reader default settings, do you know if there is a standard mode that the 508 audits use to test?
- I was not able to find a way to add that additional description without it announcing two versions of it (the form element, then the text element). Do you see that issue on the USWDS markup with the description as well? If so, is this bug something we should have them look at instead, then incorporate those changes into VADS?

Can you let us know your thoughts?

Thanks!

**Comment by coforma-jamie** [comment]
Hi @rsmithadhoc :)

My JAWS is set to automatically enter forms mode. But for whatever reason, I don't actually seem to enter it when encountering a radio button / checkbox until I activate the radio button / checkbox (by checking it off). When I TAB to a radio button group, it'll stay in browse mode until I activate one of the inputs.

That being said, I tried accessing radio buttons via USWDS and VADS and noted the behavior:

### USWDS
https://github.com/user-attachments/assets/8fd488ba-a5d1-4d97-b83e-52b998a92eb9

- radio button group --- announces the input and label while arrowing down through the list
- radio button tile group --- only announces the input and label when there's a description in the tile

### VADS
https://github.com/user-attachments/assets/012cc9ed-484b-4b81-a609-2904ae5f2615

- same behavior for both groups (standard and tile) - announces input and label while arrowing down

I hope this explanation and clips are helpful!

**Comment by amyleadem** [comment]
## Update
This appears to be an active issue isolated to certain modes in JAWS. I was not able to test on my machine because I do not yet have access to JAWS, but it appears to be active based on timeline alone. This issue was reported after the [radio a11y re-write PR](https://github.com/department-of-veterans-affairs/component-library/pull/1442) was merged. 

@jeana-adhoc, reassigning to you to confirm the issue is still active in JAWS and assess next steps.

**Comment by jeana-adhoc** [comment]
@amyleadem - Yes - this still seems to be an issue - but I'm also getting different (and better) behavior in Edge + JAWS. I'm going to work on a write up with more screens/browser/SR combinations. But for now I was able to recreate the issue still.

**Comment by jeana-adhoc** [comment]
On 6/30/25 - I tested this again in Edge, Chrome and Firefox with EDGE and ran into no issues. I was able to search for a facility, tab to move focus from the search results listing, into the first radio, and then arrow down and up.  I tried this in JAWS 2025, and 2024. In JAWS 2023 I did encounter the same issues. 

@coforma-jamie - Can you test this again and tell me if you're still encountering the same issue, and tell me which versions of JAWS you tried.  Thanks.

**Comment by hdjustice** [comment]
Thank you all so much for working through this one, so glad we didn't need to do anything in the end!
